styx-scheduler-service/src/main/java/com/spotify/styx/state/StateManager.java

```diff
@@ -58,6 +59,11 @@
   void receive(Event event) throws IsClosed;
 
   /**
+   * Get a map of all active {@link WorkflowInstance} states.
+   */
+  ImmutableMap<WorkflowInstance, RunState> activeStates();
```

Comment:
Why ImmutableMap and not just Map? Would anyone think they can mutate the return value of this and somehow that would be persisted?